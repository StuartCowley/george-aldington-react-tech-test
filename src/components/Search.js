import React, { useState } from "react";
import PropTypes from "prop-types";
import getImages from "../requests/getImages";
import { TfiSearch } from "react-icons/tfi";
import "../styles/search.css";

function Search({ setSearchResults, setLoading }) {
  const [searchValue, setSearchValue] = useState("");
  const handleSubmit = async (event) => {
    event.preventDefault();
    setSearchResults(await getImages(searchValue));
    await setLoading(false);
  };

  return (
    <>
      <form className="search-form" onSubmit={(event) => handleSubmit(event)}>
        <input
          onChange={(e) => setSearchValue(e.target.value)}
          className="search-input"
          type="text"
          data-testid="search-input"
        />
        <button
          className="search-button"
          type="submit"
          data-testid="search-button"
        >
          <TfiSearch />
        </button>
      </form>
    </>
  );
}

Search.propTypes = {
  setSearchResults: PropTypes.func.isRequired,
  setLoading: PropTypes.func.isRequired,
};

export default Search;
