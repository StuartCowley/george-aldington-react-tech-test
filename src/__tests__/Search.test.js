import { render, screen, fireEvent } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  const setSearchResults = jest.fn();
  const setLoading = jest.fn();

  it("renders Search component", () => {
    const { asFragment } = render(
      <Search setSearchResults={setSearchResults} setLoading={setLoading} />
    );
    const inputElement = screen.getByTestId("search-input");
    const buttonElement = screen.getByTestId("search-button");

    expect(asFragment()).toMatchSnapshot();
    expect(inputElement).toHaveAttribute("class", "search-input");
    expect(buttonElement).toHaveAttribute("class", "search-button");
  });

  it("calls correct function by clicking button", async () => {
    render(
      <Search setSearchResults={setSearchResults} setLoading={setLoading} />
    );
    await fireEvent.click(screen.getByTestId("search-button"));

    expect(setSearchResults).toHaveBeenCalled();
  });
});
