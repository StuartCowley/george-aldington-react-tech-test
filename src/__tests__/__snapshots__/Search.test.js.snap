// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders Search component 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      class="search-input"
      data-testid="search-input"
      type="text"
    />
    <button
      class="search-button"
      data-testid="search-button"
      type="submit"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        version="1.1"
        viewBox="0 0 17 17"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g />
        <path
          d="M16.604 15.868l-5.173-5.173c0.975-1.137 1.569-2.611 1.569-4.223 0-3.584-2.916-6.5-6.5-6.5-1.736 0-3.369 0.676-4.598 1.903-1.227 1.228-1.903 2.861-1.902 4.597 0 3.584 2.916 6.5 6.5 6.5 1.612 0 3.087-0.594 4.224-1.569l5.173 5.173 0.707-0.708zM6.5 11.972c-3.032 0-5.5-2.467-5.5-5.5-0.001-1.47 0.571-2.851 1.61-3.889 1.038-1.039 2.42-1.611 3.89-1.611 3.032 0 5.5 2.467 5.5 5.5 0 3.032-2.468 5.5-5.5 5.5z"
        />
      </svg>
    </button>
  </form>
</DocumentFragment>
`;
